src/Bicep.Core/TypeSystem/Az/IAzResourceTypeLoaderFactory.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Bicep.Core.Features;
+using Bicep.Core.Syntax;
+
+namespace Bicep.Core.TypeSystem.Az
+{
+    public interface IAzResourceTypeLoaderFactory
+    {
+        IAzResourceTypeLoader? GetResourceTypeLoader(ImportDeclarationSyntax? importDeclarationSyntax, IFeatureProvider features);
```

Comment:
Doesn't seem like syntax tree should be exposed here, why should a type loader know anything about syntax trees?  Can you just pass in the specific data that is needed?

Reply:
Especially if IAzResourceTypeLoaderFactory will be implemented by extensibility authors.

Reply:
I gues the syntax object has all the information I need, basically I need access to the resource type loader name and version. What are you proposing would be better?